service/worker/batcher/workflow_test.go

```diff
@@ -0,0 +1,275 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package batcher
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/opentracing/opentracing-go"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/suite"
+	"github.com/uber-go/tally"
+	"go.uber.org/cadence/testsuite"
+	"go.uber.org/cadence/worker"
+
+	"github.com/uber/cadence/common"
+	"github.com/uber/cadence/common/metrics"
+	mmocks "github.com/uber/cadence/common/metrics/mocks"
+	"github.com/uber/cadence/common/types"
+)
+
+type workflowSuite struct {
+	suite.Suite
+	testsuite.WorkflowTestSuite
+	workflowEnv *testsuite.TestWorkflowEnvironment
+	activityEnv *testsuite.TestActivityEnvironment
+}
+
+func TestWorkflowSuite(t *testing.T) {
+	suite.Run(t, new(workflowSuite))
```

Comment:
Nit: where possible I'd suggest moving away from the suite, since it can be fairly hard to track down or refactor changes to test data (among other things) and is obviated by most table-testing, but where we're extending existing code it totally makes sense to use.